### PR TITLE
[CA-155888] donot clean memory reservation in transfer

### PR DESCRIPTION
**Pull request by Chunjie**
Signed-off-by: Chunjie Zhu <chunjie.zhu@citrix.com>

### CODE REVIEW

**Comment by Chunjie** [comment]
add @djs55 to review

File: src/memory_server.ml [hunk 1]
```diff
@@ -132,15 +132,29 @@ let transfer_reservation_to_domain _ dbg session_id reservation_id domid =
 				let kib = Client.immediate (get_client ()) (fun xs -> Client.read xs (reservation_path _service session_id reservation_id)) in
 				(* This code is single-threaded, no need to make this transactional: *)
 				Client.immediate (get_client ()) (fun xs -> Client.write xs (Printf.sprintf "/local/domain/%d/memory/initial-reservation" domid) kib);
+                                Client.immediate (get_client ()) (fun xs -> Client.write xs (Printf.sprintf "/local/domain/%d/memory/reservation-id" domid) reservation_id);
 				Opt.iter
 					(fun maxmem -> Squeeze_xen.Domain.set_maxmem_noexn xc domid maxmem)
 					(try Some (Int64.of_string kib) with _ -> None);
-				del_reservation _service session_id reservation_id;
+				(* del_reservation _service session_id reservation_id; *)
```
Comment (djs55) [inline src/memory_server.ml]: Please delete the line rather than comment it out.

**Comment by djs55** [comment]
Apart from that minor comment, this is fine.

**Comment by Chunjie** [comment]
submit a new patch to delete the useless statement.

**Comment by jonludlam** [comment]
Have consulted with @djs55 - the with_intf shouldn't be there, nor should it be on any other functions that don't explicitly use the 'xc' parameter. Could you please remove the call from your patch, and if you're feeling particularly generous, perhaps you could also submit some further patches to remove the call from the other places you pointed out. Thanks!

**Comment by Chunjie** [comment]
Hi @jonludlam,

I change my patch according to your comments. Please see the latest commit.

In order to track other useless xenctrl.with_intf function calls, I file CA-166074, and I will submit further patches soon.

**Comment by jonludlam** [comment]
Great, thanks!

**Pull request merged**